x-pack/plugins/actions/server/saved_objects/dashboard/dashboard.ts

```diff
@@ -0,0 +1,37 @@
dda394"],"sampling":1,"incompleteColumns":{}}}},"textBased":{"layers":{}}},"internalReferences":[{"type":"index-pattern","id":"8be46dc6-05b2-4164-af01-08615858bc36","name":"indexpattern-datasource-layer-475e8ca0-e78e-454a-8597-a5492f70dce3"}],"adHocDataViews":{"8be46dc6-05b2-4164-af01-08615858bc36":{"id":"8be46dc6-05b2-4164-af01-08615858bc36","title":".kibana-event-log-*","timeFieldName":"@timestamp","sourceFilters":[],"fieldFormats":{},"runtimeFieldMap":{},"fieldAttrs":{},"allowNoIndex":false,"name":"Event Log"}}}},"hidePanelTitles":false,"enhancements":{}},"title":"Prompt + Completion Tokens per User"},{"version":"8.9.0","type":"lens","gridData":{"x":0,"y":20,"w":48,"h":20,"i":"80f745c6-a18b-492b-bacf-4a2499a2f95d"},"panelIndex":"80f745c6-a18b-492b-bacf-4a2499a2f95d","embeddableConfig":{"attributes":{"title":"","description":"","visualizationType":"lnsXY","type":"lens","references":[],"state":{"visualization":{"title":"Empty XY chart","legend":{"isVisible":true,"position":"right"},"valueLabels":"hide","preferredSeriesType":"bar_stacked","layers":[{"layerId":"475e8ca0-e78e-454a-8597-a5492f70dce3","accessors":["b0e390e4-d754-4eb4-9fcc-4347dadda394"],"position":"top","seriesType":"bar_stacked","showGridlines":false,"layerType":"data","xAccessor":"5352fcb2-7b8e-4b5a-bce9-73a7f3b2b519","yConfig":[{"forAccessor":"b0e390e4-d754-4eb4-9fcc-4347dadda394","color":"#332182"}]}]},"query":{"query":"kibana.saved_objects: { type_id  : \\".gen-ai\\" } ","language":"kuery"},"filters":[],"datasourceStates":{"formBased":{"layers":{"475e8ca0-e78e-454a-8597-a5492f70dce3":{"columns":{"5352fcb2-7b8e-4b5a-bce9-73a7f3b2b519":{"label":"user.name","dataType":"string","operationType":"terms","scale":"ordinal","sourceField":"user.name","isBucketed":true,"params":{"size":10000,"orderBy":{"type":"column","columnId":"b0e390e4-d754-4eb4-9fcc-4347dadda394"},"orderDirection":"desc","otherBucket":true,"missingBucket":false,"parentFormat":{"id":"terms"},"include":[],"exclude":[],"includeIsRegex":false,"excludeIsRegex":false},"customLabel":true},"b0e390e4-d754-4eb4-9fcc-4347dadda394":{"label":"Sum of GenAI Total Tokens","dataType":"number","operationType":"sum","sourceField":"kibana.action.execution.gen_ai.usage.total_tokens","isBucketed":false,"scale":"ratio","params":{"emptyAsNull":true},"customLabel":true}},"columnOrder":["5352fcb2-7b8e-4b5a-bce9-73a7f3b2b519","b0e390e4-d754-4eb4-9fcc-4347dadda394"],"sampling":1,"incompleteColumns":{}}}},"textBased":{"layers":{}}},"internalReferences":[{"type":"index-pattern","id":"2e702f62-8b69-41b3-b2b4-ef1798ac4632","name":"indexpattern-datasource-layer-475e8ca0-e78e-454a-8597-a5492f70dce3"}],"adHocDataViews":{"2e702f62-8b69-41b3-b2b4-ef1798ac4632":{"id":"2e702f62-8b69-41b3-b2b4-ef1798ac4632","title":".kibana-event-log-*","timeFieldName":"@timestamp","sourceFilters":[],"fieldFormats":{},"runtimeFieldMap":{},"fieldAttrs":{},"allowNoIndex":false,"name":"Event Log"}}}},"hidePanelTitles":false,"enhancements":{}},"title":"Total Tokens per User"}]',
```

Comment:
```suggestion
      JSON.stringify([
      {
        version: '8.9.0',
        type: 'lens',
        gridData: {
          x: 0,
          y: 0,
          w: 48,
          h: 20,
          i: '1e45fe29-05d3-4dbd-a0bb-fc3bc5ee3d6d',
        },
        panelIndex: '1e45fe29-05d3-4dbd-a0bb-fc3bc5ee3d6d',
        embeddableConfig: {
          attributes: {
            title: '',
            description: '',
            visualizationType: 'lnsXY',
            type: 'lens',
            references: [],
            state: {
              visualization: {
                title: 'Empty XY chart',
                legend: {
                  isVisible: true,
                  position: 'right',
                },
                valueLabels: 'hide',
                preferredSeriesType: 'bar_stacked',
                layers: [
                  {
                    layerId: '475e8ca0-e78e-454a-8597-a5492f70dce3',
                    accessors: [
                      '0f9814ec-0964-4efa-93a3-c7f173df2483',
                      'b0e390e4-d754-4eb4-9fcc-4347dadda394',
                    ],
                    position: 'top',
                    seriesType: 'bar_stacked',
                    showGridlines: false,
                    layerType: 'data',
                    xAccessor: '5352fcb2-7b8e-4b5a-bce9-73a7f3b2b519',
                    yConfig: [
                      {
                        forAccessor: '0f9814ec-0964-4efa-93a3-c7f173df2483',
                        color: '#9170b8',
                      },
                      {
                        forAccessor: 'b0e390e4-d754-4eb4-9fcc-4347dadda394',
                        color: '#3383cd',
                      },
                    ],
                  },
                ],
                labelsOrientation: {
                  x: 0,
                  yLeft: 0,
                  yRight: 0,
                },
                yTitle: 'Sum of GenAi Completion + Prompt Tokens',
                axisTitlesVisibilitySettings: {
                  x: true,
                  yLeft: true,
                  yRight: true,
                },
              },
              query: {
                query: 'kibana.saved_objects:{ type_id: ".gen-ai"   }',
                language: 'kuery',
              },
              filters: [],
              datasourceStates: {
                formBased: {
                  layers: {
                    '475e8ca0-e78e-454a-8597-a5492f70dce3': {
                      columns: {
                        '0f9814ec-0964-4efa-93a3-c7f173df2483': {
                          label: 'GenAI Completion Tokens',
                          dataType: 'number',
                          operationType: 'sum',
                          sourceField: 'kibana.action.execution.gen_ai.usage.completion_tokens',
                          isBucketed: false,
                          scale: 'ratio',
                          params: {
                            emptyAsNull: true,
                          },
                          customLabel: true,
                        },
                        '5352fcb2-7b8e-4b5a-bce9-73a7f3b2b519': {
                          label: 'user.name',
                          dataType: 'string',
                          operationType: 'terms',
                          scale: 'ordinal',
                          sourceField: 'user.name',
                          isBucketed: true,
                          params: {
                            size: 10000,
                            orderBy: {
                              type: 'custom',
                            },
                            orderDirection: 'desc',
                            otherBucket: true,
                            missingBucket: false,
                            parentFormat: {
                              id: 'terms',
                            },
                            include: [],
                            exclude: [],
                            includeIsRegex: false,
                            excludeIsRegex: false,
                            orderAgg: {
                              label: 'Sum of kibana.action.execution.gen_ai.usage.total_tokens',
                              dataType: 'number',
                              operationType: 'sum',
                              sourceField: 'kibana.action.execution.gen_ai.usage.total_tokens',
                              isBucketed: false,
                              scale: 'ratio',
                              params: {
                                emptyAsNull: true,
                              },
                            },
                            secondaryFields: [],
                          },
                          customLabel: true,
                        },
                        'b0e390e4-d754-4eb4-9fcc-4347dadda394': {
                          label: 'GenAi Prompt Tokens',
                          dataType: 'number',
                          operationType: 'sum',
                          sourceField: 'kibana.action.execution.gen_ai.usage.prompt_tokens',
                          isBucketed: false,
                          scale: 'ratio',
                          params: {
                            emptyAsNull: true,
                          },
                          customLabel: true,
                        },
                      },
                      columnOrder: [
                        '5352fcb2-7b8e-4b5a-bce9-73a7f3b2b519',
                        '0f9814ec-0964-4efa-93a3-c7f173df2483',
                        'b0e390e4-d754-4eb4-9fcc-4347dadda394',
                      ],
                      sampling: 1,
                      incompleteColumns: {},
                    },
                  },
                },
                textBased: {
                  layers: {},
                },
              },
              internalReferences: [
                {
                  type: 'index-pattern',
                  id: internalReferenceIdMapping.Tokens,
                  name: 'indexpattern-datasource-layer-475e8ca0-e78e-454a-8597-a5492f70dce3',
                },
              ],
              adHocDataViews: {
                [internalReferenceIdMapping.Tokens]: {
                  id: internalReferenceIdMapping.Tokens,
                  title: '.kibana-event-log-*',
                  timeFieldName: '@timestamp',
                  sourceFilters: [],
                  fieldFormats: {},
                  runtimeFieldMap: {
                    'kibana.action.execution.gen_ai.usage.completion_tokens': {
                      type: 'long',
                    },
                    'kibana.action.execution.gen_ai.usage.prompt_tokens': {
                      type: 'long',
                    },
                  },
                  fieldAttrs: {},
                  allowNoIndex: false,
                  name: 'Event Log',
                },
              },
            },
          },
          hidePanelTitles: false,
          enhancements: {},
        },
        title: 'Prompt + Completion Tokens per User',
      },
      {
        version: '8.9.0',
        type: 'lens',
        gridData: {
          x: 0,
          y: 20,
          w: 48,
          h: 20,
          i: '80f745c6-a18b-492b-bacf-4a2499a2f95d',
        },
        panelIndex: '80f745c6-a18b-492b-bacf-4a2499a2f95d',
        embeddableConfig: {
          attributes: {
            title: '',
            description: '',
            visualizationType: 'lnsXY',
            type: 'lens',
            references: [],
            state: {
              visualization: {
                title: 'Empty XY chart',
                legend: {
                  isVisible: true,
                  position: 'right',
                },
                valueLabels: 'hide',
                preferredSeriesType: 'bar_stacked',
                layers: [
                  {
                    layerId: '475e8ca0-e78e-454a-8597-a5492f70dce3',
                    accessors: ['b0e390e4-d754-4eb4-9fcc-4347dadda394'],
                    position: 'top',
                    seriesType: 'bar_stacked',
                    showGridlines: false,
                    layerType: 'data',
                    xAccessor: '5352fcb2-7b8e-4b5a-bce9-73a7f3b2b519',
                    yConfig: [
                      {
                        forAccessor: 'b0e390e4-d754-4eb4-9fcc-4347dadda394',
                        color: '#332182',
                      },
                    ],
                  },
                ],
              },
              query: {
                query: 'kibana.saved_objects: { type_id  : ".gen-ai" } ',
                language: 'kuery',
              },
              filters: [],
              datasourceStates: {
                formBased: {
                  layers: {
                    '475e8ca0-e78e-454a-8597-a5492f70dce3': {
                      columns: {
                        '5352fcb2-7b8e-4b5a-bce9-73a7f3b2b519': {
                          label: 'user.name',
                          dataType: 'string',
                          operationType: 'terms',
                          scale: 'ordinal',
                          sourceField: 'user.name',
                          isBucketed: true,
                          params: {
                            size: 10000,
                            orderBy: {
                              type: 'column',
                              columnId: 'b0e390e4-d754-4eb4-9fcc-4347dadda394',
                            },
                            orderDirection: 'desc',
                            otherBucket: true,
                            missingBucket: false,
                            parentFormat: {
                              id: 'terms',
                            },
                            include: [],
                            exclude: [],
                            includeIsRegex: false,
                            excludeIsRegex: false,
                          },
                          customLabel: true,
                        },
                        'b0e390e4-d754-4eb4-9fcc-4347dadda394': {
                          label: 'Sum of GenAI Total Tokens',
                          dataType: 'number',
                          operationType: 'sum',
                          sourceField: 'kibana.action.execution.gen_ai.usage.total_tokens',
                          isBucketed: false,
                          scale: 'ratio',
                          params: {
                            emptyAsNull: true,
                          },
                          customLabel: true,
                        },
                      },
                      columnOrder: [
                        '5352fcb2-7b8e-4b5a-bce9-73a7f3b2b519',
                        'b0e390e4-d754-4eb4-9fcc-4347dadda394',
                      ],
                      sampling: 1,
                      incompleteColumns: {},
                    },
                  },
                },
                textBased: {
                  layers: {},
                },
              },
              internalReferences: [
                {
                  type: 'index-pattern',
                  id: internalReferenceIdMapping.totalTokens,
                  name: 'indexpattern-datasource-layer-475e8ca0-e78e-454a-8597-a5492f70dce3',
                },
              ],
              adHocDataViews: {
                [internalReferenceIdMapping.totalTokens]: {
                  id: internalReferenceIdMapping.totalTokens,
                  title: '.kibana-event-log-*',
                  timeFieldName: '@timestamp',
                  sourceFilters: [],
                  fieldFormats: {},
                  runtimeFieldMap: {
                    'kibana.action.execution.gen_ai.usage.total_tokens': {
                      type: 'long',
                    },
                  },
                  fieldAttrs: {},
                  allowNoIndex: false,
                  name: 'Event Log',
                },
              },
            },
          },
          hidePanelTitles: false,
          enhancements: {},
        },
        title: 'Total Tokens per User',
      },
    ]),
```

Reply:
I found there's no `kibana.action.execution.gen_ai.usage.total_tokens`, `kibana.action.execution.gen_ai.usage.completion_tokens` and `kibana.action.execution.gen_ai.usage.prompt_tokens` in .kibana-event-log-* mappings, in this case, Lens wouldn't know what their types are, therefor had problems aggregating on them. Therefore I added some runtime fields for these fields so Lens can visualise them properly.

I also found when switching space, it had some conflicts, so parameterised the ids.

Reply:
@angorayc did you start a new instance? If the event log was created before the [mappings PR](https://github.com/elastic/kibana/pull/158508) was merged, that would make sense it is missing the mappings